### PR TITLE
records: add geolocation support

### PR DESCRIPTION
**Pull request by pazembrz**
* Adds mixin which tries to find Conferences and Institutions latitude and longitude based on data provided in 'addresses' field

### CODE REVIEW

File: backend/inspirehep/records/api/mixins.py [hunk 1]
```diff
@@ -637,3 +637,59 @@ def get_modified_experiment_uuids(self):
...
+                address_list = [place_name, postal_code, city, state, country]
+                postal_address = ", ".join(
+                    [addr_part for addr_part in address_list if addr_part]
+                )
+            LOGGER.info(f"Querying OSM for address: {postal_address}")
+            latitude, longitude = cls.get_geolocation(postal_address)
+            if latitude and longitude:
+                LOGGER.info("Geolocation data has been found.")
+                data["addresses"][idx]["latitude"] = latitude
+                data["addresses"][idx]["longitude"] = longitude
```
Comment (michamos) [inline backend/inspirehep/records/api/mixins.py:694]:
```suggestion
                address["latitude"] = latitude
                address["longitude"] = longitude
```
so you don't need to `enumerate`.

File: backend/inspirehep/records/api/mixins.py [hunk 2]
```diff
@@ -637,3 +637,59 @@ def get_modified_experiment_uuids(self):
...
+
+    @classmethod
+    def build_address(cls, data):
+        for idx, address in enumerate(data.get("addresses", [])):
+            latitude = address.get("latitude")
+            longitude = address.get("longitude")
+            if latitude and longitude:
+                LOGGER.info("Using stored geolocation data.")
+                continue
+            postal_address = address.get("postal_address")[0]
```
Comment (michamos) [inline backend/inspirehep/records/api/mixins.py:677]: `KeyError` if no `postal_address`.

File: backend/inspirehep/records/api/mixins.py (hunk 1 above)
Comment (pazembrz) [reply backend/inspirehep/records/api/mixins.py]: :+1:

File: backend/inspirehep/records/api/mixins.py (hunk 2 above)
Comment (pazembrz) [reply backend/inspirehep/records/api/mixins.py]: :+1:

**Pull request closed without merging**